lib/Dialect/TTIR/Analysis/ShardSolver.cpp

```diff
@@ -435,33 +505,31 @@ bool ShardSolver::checkShardCompatible(const Operation *producerOp,
   // currentOp output tensor shard spec, nextOp exec and nextOp output
   // tensor.
   //
-  llvm::SmallVector<int64_t> producerOpShardShape =
-      producerLayout.getShardShape(false /*convertTileToScalar*/);
-  uint64_t l1InputUsage = producerLayout.getElementSizeBytes();
-  for (int64_t dim : producerOpShardShape) {
-    l1InputUsage *= dim;
-  }
-
-  llvm::SmallVector<int64_t> consumerOpShardShape =
-      consumerLayout.getShardShape(false /*convertTileToScalar*/);
-  uint64_t l1OutputUsage = consumerLayout.getElementSizeBytes();
-  for (int64_t dim : consumerOpShardShape) {
-    l1OutputUsage *= dim;
-  }
-
+  RankedTensorType producerTensorType =
+      mlir::cast<RankedTensorType>(producerOp->getResult(0).getType());
+  uint64_t producerL1OutputUsage = deviceAttr.getLayoutSizeBytes(
+      producerTensorType.getShape(), producerLayout,
+      producerLayout.getMemorySpace());
```

Comment:
Can we assert that the memory space is in fact L1?

Reply:
`hasShardedTensorMemoryLayout` could theoretically be relaxed to support DRAM sharded, which TTNN plans to support (if they haven't already).

Reply:
Ill fork it to hasShardedL1TensorMemoryLayout.